### PR TITLE
Fixed YAML deserialization of *bool

### PR DESCRIPTION
**Pull request by anuptalwalkar**
GFM-128

### CODE REVIEW

File: core/config/yaml_test.go [hunk 1]
```diff
@@ -146,3 +146,26 @@ func TestMatchPopulatedEmptyStruct(t *testing.T) {
...
+	provider.GetValue("pointerStruct").PopulateStruct(&ps)
+	assert.True(t, *ps.MyBool)
+}
+
+func TestNonExistingBoolPointer(t *testing.T) {
+	provider := NewProviderGroup("global", NewYAMLProviderFromBytes([]byte(``)))
+	ps := pointerStruct{}
+	provider.GetValue("pointerStruct").PopulateStruct(&ps)
+	assert.Nil(t, ps.MyBool)
+}
```
Comment (sectioneight) [inline core/config/yaml_test.go]: Can you add a test with a `*string` field too? We should make sure this works for types other than just bools.

File: core/config/yaml_test.go [hunk 2]
```diff
@@ -146,3 +146,26 @@ func TestMatchPopulatedEmptyStruct(t *testing.T) {
...
+pointerStruct:
+  myBool: true
+`)
+
+func TestBoolPointer(t *testing.T) {
+	provider := NewProviderGroup("global", NewYAMLProviderFromBytes(myBoolyaml))
+	ps := pointerStruct{}
+	provider.GetValue("pointerStruct").PopulateStruct(&ps)
+	assert.True(t, *ps.MyBool)
+}
```
Comment (sectioneight) [inline core/config/yaml_test.go]: for completeness sake, can we also check an explicit `false`? it'd be a shame if we messed that up

File: core/config/yaml_test.go (hunk 1 above)
Comment (anuptalwalkar) [inline core/config/yaml_test.go]: Sure. Adding it.

File: core/config/yaml_test.go [hunk 3]
```diff
@@ -146,3 +146,34 @@ func TestMatchPopulatedEmptyStruct(t *testing.T) {
...
+	MyTrueBool  *bool   `yaml:"myTrueBool"`
+	MyFalseBool *bool   `yaml:"myFalseBool"`
+	MyString    *string `yaml:"myString"`
+}
+
+var pointerYaml = []byte(`
+pointerStruct:
+  myTrueBool: true
+  myFalseBool: false
+  myString: "hello"
```
Comment (sectioneight) [inline core/config/yaml_test.go]: nit: no need to surround "hello" with quotes in YAML

File: core/config/yaml_test.go [hunk 4]
```diff
@@ -146,3 +146,34 @@ func TestMatchPopulatedEmptyStruct(t *testing.T) {
 	empty := reflect.New(reflect.TypeOf(es)).Elem().Interface()
 	assert.True(t, reflect.DeepEqual(empty, es))
 }
+
+type pointerStruct struct {
+	MyTrueBool  *bool   `yaml:"myTrueBool"`
+	MyFalseBool *bool   `yaml:"myFalseBool"`
```
Comment (glibsm) [inline core/config/yaml_test.go]: Shouldn't we add `MyMissingBool`, which is not present in YAML, so we can make sure that stays nil?

Comment (anuptalwalkar) [inline core/config/yaml_test.go]: Yes, there is a separate test for checking nil *bools and other types on empty yaml.

**Pull request merged**